src/OrchardCore/OrchardCore/Localization/DefaultCalendarManager.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using NodaTime;
+
+namespace OrchardCore.Localization
+{
+    public class DefaultCalendarManager : ICalendarManager
+    {
+        private readonly ICalendarSelector _calendarSelector;
+
+        private static IDictionary<string, CalendarSystem> CalendarsList = CalendarSystem.Ids
+            .ToLookup(c => CalendarSystem.ForId(c).Name)
+            .ToDictionary(c => c.Key, c => CalendarSystem.ForId(c.First()));
+
+        public DefaultCalendarManager(ICalendarSelector calendarSelector)
```

Comment:
The pattern we use when we do selectors is to inject an `IEnumerable<ICalendarSelector>` then sort all the results that are not null by `Priority` and take the first one. You can see btw that you are not using `Priority` anywhere.